### PR TITLE
<MessageBoxMarketerialLayout/> - primaryButtonNode

### PR DESCRIPTION
**Pull request by yogevitz**
### 🔦 Summary

Added the option to assign a node prop for the MessageBoxMarketerialLayout component, which will replace the hardcoded primary button.
Upon using primaryButtonNode, the regular primaryButton will not be shown.
This allows adding a custom primary button for the MessageBoxMarketerialLayout.
Added an example to the storybook using a Button inside a DropDownBase component.

This PR is replacing the old PR: https://github.com/wix/wix-style-react/pull/4072

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: src/MessageBox/MarketerialLayout/MessageBoxMarketerialLayout.js [hunk 1]
```diff
@@ -92,7 +93,10 @@ class MessageBoxMarketerialLayout extends WixComponent {
     } = this.props;
     return (
       <div className={styles.buttonsContainer}>
-        {primaryButtonLabel ? (
+        {!!primaryButtonNode ? (
+          <div data-hook="primary-button-node">{primaryButtonNode}</div>
+        ) : null}
```
Comment (nitayneeman) [inline src/MessageBox/MarketerialLayout/MessageBoxMarketerialLayout.js]: Minor, but could be just:
```jsx
{!!primaryButtonNode && (<div data-hook="primary-button-node">{primaryButtonNode}</div>)}
```

File: src/MessageBox/MarketerialLayout/MessageBoxMarketerialLayout.spec.js [hunk 2]
```diff
@@ -67,6 +68,32 @@ describe('MessageBoxMarketerialLayout', () => {
...
+        const props = Object.assign(
+          {},
+          requiredProps,
+          {
+            primaryButtonNode: <Button>Hi</Button>,
+          },
+          {},
+        );
+        const driver = createDriver(<MessageBoxMarketerialLayout {...props} />);
+        expect(await driver.getPrimaryButtonNode()).toBeTruthy();
```
Comment (nitayneeman) [inline src/MessageBox/MarketerialLayout/MessageBoxMarketerialLayout.spec.js]: Also minor, but we prefer:
```suggestion
        expect(await driver.getPrimaryButtonNode()).toBe(true);
```

Because of:
https://vincenttunru.com/toBeTruthy-vs-toBe-true/

**Comment by nitayneeman** [comment]
We should fix the broken unit tests and also the types checking.

**Comment by shlomitc** [comment]
@nitayneeman @yogevitz - any update grading this PR?

File: src/MessageBox/docs/AnnouncementExamples/PrimaryButtonNode.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { MessageBoxMarketerialLayout } from 'wix-style-react/MessageBox';
+import DropdownBase from 'wix-style-react/DropdownBase';
+import Button from 'wix-style-react/Button';
+import ChevronDown from 'wix-style-react/new-icons/ChevronDown';
```
Comment (nitayneeman) [inline src/MessageBox/docs/AnnouncementExamples/PrimaryButtonNode.js]: Replace with:
```suggestion
import ChevronDown from 'wix-ui-icons-common/ChevronDown';
```

I think it will solve this issue:
```
Module not found: Error: Can't resolve 'wix-style-react/new-icons/ChevronDown' in '/home/builduser/agent00/work/dbb751eb2b5c5b2e/src/MessageBox/docs/AnnouncementExamples'
```

File: src/MessageBox/MarketerialLayout/MessageBoxMarketerialLayout.driver.js [hunk 4]
```diff
@@ -11,6 +13,7 @@ const messageBoxMarketerialLayoutDriverFactory = ({ element }) => {
     exists: () => !!element,
     getPrimaryButtonText: () => primaryButton().textContent,
     getPrimaryButton: primaryButton,
+    getPrimaryButtonNode: primaryButtonNode,
```
[the remark this answers is not in the document]
Comment (yogevitz) [reply src/MessageBox/MarketerialLayout/MessageBoxMarketerialLayout.driver.js]: done

File: package.json [hunk 5]
```diff
@@ -194,7 +194,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.3.1",
-    "@loadable/component": "^5.10.3",
+    "@loadable/component": "^5.12.0",
```
Comment (nitayneeman) [inline package.json]: I would revert this change unless you've a decent reason :)
